the code is already nicely written to use _exit() and execve(), so just
using vfork() over fork() seems to work fine.

--- boa-0.94.14rc21/src/cgi.c
+++ boa-0.94.14rc21/src/cgi.c
@@ -445,7 +445,7 @@
         }
     }
 
-    child_pid = fork();
+    child_pid = vfork();
     switch (child_pid) {
     case -1:
         /* fork unsuccessful */
--- boa-0.94.14rc21/src/sublog.c
+++ boa-0.94.14rc21/src/sublog.c
@@ -48,7 +48,7 @@
      * filedes[1] is for writing. */
     if (pipe(pipe_fds) == -1)
         return -1;
-    pid = fork();
+    pid = vfork();
     if (pid == 0) {             /* child */
         close(pipe_fds[1]);
         if (pipe_fds[0] != 0) {
@@ -56,7 +56,7 @@
             close(pipe_fds[0]);
         }
         execl("/bin/sh", "sh", "-c", command, (char *) 0);
-        exit(EXIT_FAILURE);
+        _exit(EXIT_FAILURE);
     }
     close(pipe_fds[0]);
     if (pid < 0) {
